### PR TITLE
Standalone packs for Sega consoles: Game Gear, Genesis, SMS.

### PR DESCRIPTION
**Pull request by hotelvictorcharlie**
This is along the same lines as #846.

I think there are many benefits to creating standalone packs for these Sega consoles, because they are supported by many devices (such as various EverDrives, MiSTer, Nt Mini, and Analogue Pocket). 

SG-1000 has [already been split out into a standalone pack](https://github.com/frederic-mahe/Hardware-Target-Game-Database/pull/483), and I think we should do the same thing with these other consoles for similar reasons. Right now we have a `Master EverDrive` pack which includes `Sega SG-1000` via a placeholder. And we have other packs that include `Master EverDrive` as a placeholder, which creates two levels of nesting. While some devices that support SMS roms also support SG-1000 roms, some (such as the EverDrive GG and Mega EverDrive) do not, and others (such as the Analogue Pocket, which I want to make a pack for) expect the SG-1000 roms to live outside the parent SMS directory. Therefore, I think we should get rid of the placeholders that point to `Master EverDrive` because not all devices support the nested placeholders that it references (and even on the ones that do, sometimes they want those files to be in a different location).

The same is also true of the other consoles. For example, the EverDrive GG supports SMS roms, but right now our `EverDrive GG` pack doesn't contain any. So we want to add a placeholder. But the device *does not* support SG-1000 roms. So we should have a placeholder in the `EverDrive GG` pack that can point to a `Master System` pack, but not the `SG-1000` pack. This PR fixes that.

And while the Mega EverDrive supports 32X roms, the devices that refer to the `Mega EverDrive` pack via placeholders (Genesis_MiSTer, Nt Mini Noir, and now Analogue Pocket) do not, so it doesn't make much sense for us to be referencing them from other packs designed for devices that can't run them.

Also, this solves a lot of the existing duplication between the Master EverDrive and Analogue Nt Mini packs, and fixes the problem of them getting out of sync (for example, [this update from 4 months ago](https://github.com/frederic-mahe/Hardware-Target-Game-Database/commit/04ba094361b0a8adde78e10ebe38d15dbe7e2009) still hasn't been applied to the SMS roms inside the Nt Mini packs).

I strongly suggest doing a "Squash and Merge" on this PR due to the large number of commits.

### CODE REVIEW

File: EverDrive Pack SMDBs/MiSTer.txt [hunk 1]
```diff
@@ -1296,7 +1296,7 @@ ad5cdec55d95106094d1b822dc36d9e9c79616d60877c1c10b4c15eb7e35b25a	MiSTer/Games/Ga
 eccd2105f74efbd0998c13a15424c21ef07064512c5a1bf32f97f25dabce8607	MiSTer/Games/Galaksija/ZvjezdaneStaze.tap	9a97ec2cb2875d0bee2d0f175f9051ee3c4a1f21	2ac26f35acb182fa847ff55fa0bd86c2	61b6cee8
 3971d656ecfb142d0c5d639ea491dd70b2ec1f98df9efc2cdfd2b442a68c2676	MiSTer/Games/Gameboy/boot0.rom	040b7a5583e4764a04e1cdb05d409add949a98f2	10d4b62e39a0c47615770ffab949d0b1	0ce4a2c7
 b4f2e416a35eef52cba161b159c7c8523a92594facb924b3ede0d722867c50c7	MiSTer/Games/Gameboy/boot1.rom	1293d68bf9643bc4f36954c1e80e38f39864528d	dbfce9db9deaa2567f6a84fde55f9680	41884e46
-e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855	MiSTer/Games/Genesis/@Place Sega Genesis Pack Here	da39a3ee5e6b4b0d3255bfef95601890afd80709	d41d8cd98f00b204e9800998ecf8427e	00000000
+e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855	MiSTer/Games/Genesis/@Place Genesis Pack Here	da39a3ee5e6b4b0d3255bfef95601890afd80709	d41d8cd98f00b204e9800998ecf8427e	00000000
```
Comment (hotelvictorcharlie) [inline EverDrive Pack SMDBs/MiSTer.txt:1299]: This is just for consistency in naming. Also, before this PR we didn't have a Genesis pack, so this was actually referring to the `Mega EverDrive` pack.

File: EverDrive Pack SMDBs/MiSTer.txt [hunk 2]
```diff
@@ -2109,8 +2109,9 @@ f06c977b619c1e4b9be16868c16b7ccc3049ec21ccd7f8edb7432c7dca10b903	MiSTer/Games/PD
...
-e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855	MiSTer/Games/SMS/Master System/@Place Master EverDrive Pack Here	da39a3ee5e6b4b0d3255bfef95601890afd80709	d41d8cd98f00b204e9800998ecf8427e	00000000
+e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855	MiSTer/Games/SMS/Game Gear/@Place Game Gear Pack Here	da39a3ee5e6b4b0d3255bfef95601890afd80709	d41d8cd98f00b204e9800998ecf8427e	00000000
+e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855	MiSTer/Games/SMS/Master System/@Place Master System Pack Here	da39a3ee5e6b4b0d3255bfef95601890afd80709	d41d8cd98f00b204e9800998ecf8427e	00000000
+e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855	MiSTer/Games/SMS/SG-1000/@Place Sega SG-1000 Pack Here	da39a3ee5e6b4b0d3255bfef95601890afd80709	d41d8cd98f00b204e9800998ecf8427e	00000000
 e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855	MiSTer/Games/SNES/Place SNES Pack Here	da39a3ee5e6b4b0d3255bfef95601890afd80709	d41d8cd98f00b204e9800998ecf8427e	00000000
```
Comment (hotelvictorcharlie) [inline EverDrive Pack SMDBs/MiSTer.txt:2115]: This will probably create a merge conflict with #841 but that's an easy fix.

**Comment by tjanas** [comment]
I am against this change.

Also keep in mind, Master System is not a simple drop-in for Analogue Nt Mini; it specifically requires some files to be renamed with specific file extensions to indicate specific mappers. Ideally, both Nt Mini consoles would receive new JB firmware updates to eliminate this dependency so that the same set of filenames could be used across the board, which would then permit a drop-in for Nt Mini.

I don't think there should be both a "Master EverDrive SMDB" and "Master System SMDB". They are logically the same device targeting the same hardware target.

I would like to hear thoughts from @SmokeMonsterPacks @frederic-mahe on this before making any major changes like this.

**Comment by hotelvictorcharlie** [comment]
> Also keep in mind, Master System is not a simple drop-in for Analogue Nt Mini; it specifically requires some files to be renamed with specific file extensions to indicate specific mappers. Ideally, both Nt Mini consoles would receive new JB firmware updates to eliminate this dependency so that the same set of filenames could be used across the board, which would then permit a drop-in for Nt Mini.

I don't know if we'll ever get a firmware update to fix that, but 97.5% of the roms in the Nt Mini packs' SMS directories have an `.sms` or `.sg` extension. Perhaps we could simply move the few outliers into an "NT Fixed" directory (like we do with Atari 2600 and NES)?

> I don't think there should be both a "Master EverDrive SMDB" and "Master System SMDB". They are logically the same device targeting the same hardware target.

Except they aren't, because we are currently trying to use this one pack to support several different devices, each of which either does or doesn't support SG-1000.

**Comment by hotelvictorcharlie** [comment]
> I don't know if we'll ever get a firmware update to fix that, but 97.5% of the roms in the Nt Mini packs' SMS directories have an `.sms` or `.sg` extension. Perhaps we could simply move the few outliers into an "NT Fixed" directory (like we do with Atari 2600 and NES)?

I just added this in 29d666e.

**Pull request closed without merging**